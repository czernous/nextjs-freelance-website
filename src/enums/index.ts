export enum Color {
  Olive = 'olive',
  Gray = 'gray',
  Bark = 'bark',
  Brick = 'brick',
  Danger = 'danger',
  Success = 'success',
  Warning = 'warning',
  Info = 'info',
}

export enum Size {
  Small = 'small',
  Regular = 'regular',
  Large = 'large',
  ExtraLarge = 'extra-large',
}

export enum InputTypes {
  Button = 'button',
  Checkbox = 'checkbox',
  Color = 'color',
  Date = 'date',
  Email = 'email',
  File = 'file',
  Hidden = 'hidden',
  Image = 'image',
  Month = 'month',
  Number = 'number',
  Password = 'password',
  Text = 'text',
}
