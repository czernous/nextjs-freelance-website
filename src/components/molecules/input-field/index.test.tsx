import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Color } from '../../../enums';
import InputField from './index';
import React from 'react';

describe('Plain Input', () => {
  let input;

  beforeEach(async () => {
    input = render(
      <InputField
        type="input"
        inputId="test_input"
        inputPlaceholder="Enter your name"
        inputColor={Color.Brick}
        fieldLabel={'Name'}
        fieldClasses={''}
      />,
    );
  });

  it('renders input field unchanged', () => {
    const { container } = input;
    expect(container).toMatchSnapshot();
  });

  it('renders Input Field', () => {
    const input = screen.getByPlaceholderText('Enter your name');
    expect(input).toBeInTheDocument();
  });
});
