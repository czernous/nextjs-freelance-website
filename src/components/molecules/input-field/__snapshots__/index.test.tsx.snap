// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plain Input renders input field unchanged 1`] = `
<div>
  <div
    class=" input"
  >
    <label
      for="test_input"
    >
      Name
    </label>
    <input
      class="undefined brick input"
      id="test_input"
      placeholder="Enter your name"
    />
  </div>
</div>
`;
